tpot/base.py

```diff
@@ -636,32 +631,66 @@ def _update_top_pipeline(self, features, target):
                                   'TPOTClassifier object. Please make sure you '
                                   'passed the data to TPOT correctly.')
             else:
-                self.fitted_pipeline_ = self._toolbox.compile(expr=self._optimized_pipeline)
+                if not self._last_optimized_pipeline:
+                    self._last_optimized_pipeline = str(self._optimized_pipeline)
+                elif self._last_optimized_pipeline == str(self._optimized_pipeline):
```

Comment:
Maybe the check should be whether the Pareto front has changed at all in the last *n* generations, rather than whether the single "best" pipeline has changed.